shared/actions/fs/common.native.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import * as Saga from '../../util/saga'
+import * as FsGen from '../fs-gen'
+import {navigateAppend} from '../../actions/route-tree'
+
+export const share = (action: FsGen.SharePayload) =>
+  Saga.put(
+    navigateAppend([
+      {
+        props: {path: action.payload.path, isShare: true},
+        selected: 'pathItemAction',
+      },
+    ])
+  )
+
+export function* save(action: FsGen.SavePayload): Saga.SagaGenerator<any, any> {
+  const {path} = action.payload
+  Saga.put(FsGen.createDownload({path, intent: 'camera-roll'}))
+  Saga.put(
+    navigateAppend([
```

Comment:
if you do this you won't have a protected navigateappend. you might want to use putActionIfOnPath and pass that as a payload param

Reply:
> a protected navigateappend

Is this to protect against double append?

Reply:
yes, else its just 'appending' to whatever the current path happens to be, which could be anything